algoliasearch/src/main/java/com/algolia/search/saas/BaseAPIClient.java

```diff
@@ -330,15 +349,75 @@ private String _getAnswer(InputStream istream) throws IOException {
         return builder.toString();
     }
 
+    /**
+     * Reads the InputStream into a byte array
+     * @param stream the InputStream to read
+     * @return the stream's content as a byte[]
+     * @throws AlgoliaException if the stream can't be read or flushed
+     */
+    private byte[] _toByteArray(InputStream stream) throws AlgoliaException {
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        int read;
+        byte[] buffer = new byte[1024];
+
+        try {
+            while ((read = stream.read(buffer, 0, buffer.length)) != -1) {
+                out.write(buffer, 0, read);
+            }
+
+            out.flush();
+            return out.toByteArray();
+        } catch (IOException e) {
+            throw new AlgoliaException("Error while reading stream: " + e.getMessage());
+        }
+    }
+
+
     private JSONObject _getJSONObject(String input) throws JSONException {
         return new JSONObject(new JSONTokener(input));
     }
 
-    private JSONObject _getAnswerObject(InputStream istream) throws IOException, JSONException {
-        return _getJSONObject(_getAnswer(istream));
+    private JSONObject _getJSONObject(byte[] array) throws JSONException {
+        return new JSONObject(new JSONTokener(new String(array)));
```

Comment:
- `new String(byte[])` without specifying a charset is dangerous. Although the default charset is supposed to be UTF-8 on Android, I would feel more comfortable specifying the charset explicitly.
- `JSONTokener` is superfluous as `JSONObject` can construct itself from a String.